core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -537,10 +537,19 @@ private[spark] class ExecutorAllocationManager(
       }
     }
 
+    override def onTaskResubmit(taskResubmit: SparkListenerTaskResubmit): Unit = {
+      val stageId = taskResubmit.stageId
+      allocationManager.synchronized {
+        val num = stageIdToNumTasks.getOrElse(stageId, 0)
+        stageIdToNumTasks.update(stageId, num + 1)
+      }
+    }
+
```

Comment:
@squito, I think when an executor goes down, the stages won't be resubmitted. Here I means when a task fails, it will retry, and so a new task will append. And in order to let the **ExecutorAllocationManager** knows there are new tasks submit, I add **SparkListenerTaskResubmit** event.